powerline/segments/common/env.py

```diff
@@ -166,6 +169,11 @@ def user(pl, hide_user=None):
 		return None
 	if username == hide_user:
 		return None
+	if hide_domain == True:
+		searchfor = None
+		searchfor = re.search('@', username)
+		if searchfor:
+			username, _ = username.split('@')
```

Comment:
After index is found there is absolutely no need to do the split: you should use the found index. Match object *also* contains index: `.end()` and `.start()` methods.